src/Logger_new.java

```diff
@@ -0,0 +1,340 @@
+package com.aap.gitst;
+
+import java.io.Console;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.LockSupport;
+
+/**
+ * @author Andrey Pavlenko
+ */
+public class Logger {
+    private final PrintWriter _logWriter;
+    private final List<PBar> _pbar = new ArrayList<PBar>(3);
+    private volatile Level _level;
+    private volatile boolean _progressBarSupported;
+
+    public Logger(final OutputStream logWriter,
+            final boolean progressBarSupported) {
+        this(logWriter, progressBarSupported, null);
+    }
+
+    public Logger(final OutputStream logWriter,
+            final boolean progressBarSupported, final Level level) {
+        this(new PrintWriter(logWriter), progressBarSupported, level);
+    }
+
+    public Logger(final PrintWriter logWriter,
+            final boolean progressBarSupported) {
+        this(logWriter, progressBarSupported, null);
+    }
+
+    public Logger(final PrintWriter logWriter,
+            final boolean progressBarSupported, final Level level) {
+        _logWriter = logWriter;
+        _progressBarSupported = progressBarSupported;
+
+        if (level == null) {
+            if ("true".equalsIgnoreCase(System.getenv("GITST_DEBUG"))) {
+                _level = Level.DEBUG;
+            } else {
+                _level = Level.INFO;
+            }
+        } else {
+            _level = level;
+        }
+    }
+
+    public static Logger createConsoleLogger(final Level level) {
+        final Console c = System.console();
+        final String env = System.getenv("GITST_PB");
+        final boolean enable = "true".equalsIgnoreCase(env);
+        final boolean disable = "false".equalsIgnoreCase(env);
+
+        if (c != null) {
+            return new Logger(c.writer(), !disable, level);
+        } else {
+            return new Logger(System.out, enable, level);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS106)